sql/parser/eval.go

```diff
@@ -771,40 +1166,26 @@ func evalRangeCond(expr *RangeCond) (Datum, error) {
 	// AND left <= to". The only tricky part is that we evaluate "left" only
 	// once.
 
-	left, err := EvalExpr(expr.Left)
-	if err != nil {
-		return DNull, err
-	}
-
-	limits := [2]struct {
-		op   ComparisonOp
-		expr Expr
-	}{
-		{GE, expr.From},
-		{LE, expr.To},
+	var newExpr Expr = &AndExpr{
```

Comment:
Why this change? It doesn't seem related to this type-checking. More to the point, this code shouldn't even be necessary anymore now that `NormalizeExpr` does exactly this normalization.